idl/github.com/uber/cadence/shared.thrift

```diff
@@ -550,6 +553,7 @@ struct ExternalWorkflowExecutionCancelRequestedEventAttributes {
   10: optional i64 (js.type = "Long") initiatedEventId
   20: optional string domain
   30: optional WorkflowExecution workflowExecution
+  40: optional binary control
```

Comment:
because we cannot use the workflow ID as unique identifier for workflow cancellation.
so, we must come up with a "ID", and control is the ID.
the reason i use "binary control" instead of "int64 id" is to follow the existing pattern